server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardLoopIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+            Optional.empty(),
+            Rule.builder()
+                .id(Rule.Id.of("1"))
+                .name("rule 1")
+                .conditionGroup(Rule.ConditionGroup.of(Rule.ConditionCombiner.AND, Rule.Condition.of(FROM, NOT_CONTAINS, "AAA")))
+                .action(Rule.Action.builder().setAppendInMailboxes(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of()))
+                    .setWithKeywords(ImmutableList.of())
+                    .setForward(Optional.of(Rule.Action.Forward.of(ImmutableList.of(ALICE.asMailAddress()), false)))
+                    .build())
+                .build());
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER.asString(), PASSWORD)
+            .sendMessage(SENDER.asString(), ALICE.asString());
+
+        Awaitility.waitAtMost(1L, TimeUnit.HOURS).until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 2L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            List<Mail> mails = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
+                .filter(Throwing.predicate(mail -> mail.getRecipients().contains(ALICE.asMailAddress())))
+                .collect(ImmutableList.toImmutableList());
+
+            softly.assertThat(mails.get(0).getRecipients()).containsOnly(ALICE.asMailAddress());
+            softly.assertThat(mails.get(1).getRecipients()).containsOnly(ALICE.asMailAddress());
+        }));
+    }
+
+    @Test
+    void regularForwardShouldNotCreateLoopError() throws Exception {
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(ALICE),
+                Mapping.forward(SENDER.asString()));
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(ALICE),
+                Mapping.forward(BOB.asString()));
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(BOB),
+                Mapping.forward(CEDRIC.asString()));
+        jamesServer.getProbe(DataProbeImpl.class)
+            .addMapping(MappingSource.fromUser(CEDRIC),
+                Mapping.forward(ALICE.asString()));
+
+        messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+            .authenticate(SENDER.asString(), PASSWORD)
+            .sendMessage(SENDER.asString(), ALICE.asString());
+
+//        Awaitility.await().pollDelay(1, TimeUnit.SECONDS).until(() -> jamesServer.getProbe(SpoolerProbe.class).processingFinished());
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
+
+        SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+            List<Mail> mails = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
+                .collect(ImmutableList.toImmutableList());
+
+            softly.assertThat(mails).hasSize(1);
```

Comment:
2 things:
- `Awaitility.await().until` above already checks that?
- I would think cedric gets one and sender gets one too?

Reply:
They do not keep a copy by default

Reply:
cf https://github.com/apache/james-project/pull/1758#discussion_r1368334183